demo/docs/features/errors.md

```diff
@@ -0,0 +1,53 @@
+# Errors and Exceptions
+
+All apps will eventually throw an error that will be treated by developers or exceptions when something unforeseen
+happens.
+
+## Description
+
+### Client
+
+In the case of errors (i.e. HTTP calls that return `4xx` or `5xx` status codes), we wrap the call in a `try/catch` block
+for `async/await` operations or using `then` and `catch` for promises and use the `pushError` API provided by Grafana
+JavaScript Agent to report them.
+
+Exceptions are reported automatically as Grafana JavaScript Agent offers a way to automatically capture exceptions.
```

Comment:
Could rewrite to be more direct, something like "The Grafana JavaScript Agent automatically captures exceptions". Might be useful to also include how we detect exceptions.